src/mtcp/mtcp_restart.c

```diff
@@ -519,20 +523,47 @@ setupTempRegions() {
     mtcp_abort();
   }
 
+  // In May 2020, on Cori and elsewhere, vvar is 3 pages and vdso is 2 pages.
   while (mtcp_readmapsline(mapsfd, &area)) {
     if (prev_addr + 5 * PAGESIZE <= area.addr) {
       vvarStartTmp = prev_addr;
       vdsoStartTmp = prev_addr + 3 * PAGESIZE;
       break;
     }
-    prev_addr = ROUNDADDRUP((uint64_t) area.endAddr, 0x1000);
+    prev_addr = ROUNDADDRUP((uint64_t) area.endAddr, PAGESIZE);
   }
   mtcp_sys_close(mapsfd);
 
   if (vvarStartTmp < 0 || vdsoStartTmp < 0) {
     MTCP_PRINTF("No free region found to temporarily map vvar/vdso to\n");
     mtcp_abort();
   }
+
+  // We have to iterate through the memory map twice, this time to ensure that
+  // free regions have been found for us to remap to.
+  mapsfd = mtcp_sys_open2("/proc/self/maps", O_RDONLY);
```

Comment:
One of the reasons why I suggested to combine them in the same function is because I don't think it necessary to re-open the same file and re-iterate again.